### PR TITLE
Integration specs

### PR DESCRIPTION
**Pull request by Fadahunsiseyi**
### In this MileStone, I performed the following;

Use Capybara to write integration tests for each view in your project.

User index page:

I can see the username of all other users.
I can see the profile picture for each user.
I can see the number of posts each user has written.
When I click on a user, I am redirected to that user's show page.
user show page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see the user's bio.
I can see the user's first 3 posts.
I can see a button that lets me view all of a user's posts.
When I click a user's post, it redirects me to that post's show page.
When I click to see all posts, it redirects me to the user's post's index page.
User post index page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see a post's title.
I can see some of the post's body.
I can see the first comments on a post.
I can see how many comments a post has.
I can see how many likes a post has.
I can see a section for pagination if there are more posts than fit on the view.
When I click on a post, it redirects me to that post's show page.
Post show page:

I can see the post's title.
I can see who wrote the post.
I can see how many comments it has.
I can see how many likes it has.
I can see the post body.
I can see the username of each commentor.
I can see the comment each commentor left.

### CODE REVIEW

File: spec/views/post_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  end
+
+  scenario 'can see the comments of each commentor.' do
+    expect(page).to have_content 'Nice post!!'
+    expect(page).to have_content 'Nice post!!'
+    expect(page).to have_content 'Nice post!!'
+    expect(page).to have_content 'Nice post!!'
+    expect(page).to have_content 'Nice post!!'
+  end
+end
```
Comment (Whoistolu) [inline spec/views/post_show_spec.rb:48]: - [ ] [REQUIRED] We have 4 tests failing for the `spec/requests/posts_spec.rb` and the `spec/requests/users_spec.rb:`. The `request` directory is not part of your file changes so I decided to attach them here.  Kindly fix the test errors before asking for another review.
Screenshot 2022-11-01 at 20 31 14

Screenshot 2022-11-01 at 20 37 11

Screenshot 2022-11-01 at 20 36 26

Screenshot 2022-11-01 at 20 34 52

Screenshot 2022-11-01 at 20 52 15

File: spec/views/user_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    visit '/'
+    expect(page).to have_content(@user.posts_counter)
+  end
+
+  it "redirected to that user's show page when I click on a user" do
+    visit '/'
+    click_link(@user.name)
+    expect(page).to have_content(@user.name)
+  end
+end
```
Comment (Whoistolu) [inline spec/views/user_index_spec.rb:33]: - [ ] [REQUIRED] We have 4 tests failing for the `spec/requests/posts_spec.rb` and the `spec/requests/users_spec.rb:`. The `request` directory is not part of your file changes so I decided to attach them here.  Kindly fix the test errors before asking for another review.
Screenshot 2022-11-01 at 20 31 14

Screenshot 2022-11-01 at 20 37 11

Screenshot 2022-11-01 at 20 36 26

Screenshot 2022-11-01 at 20 34 52

Screenshot 2022-11-01 at 20 52 15

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @Fadahunsiseyi 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/user_index_spec.rb (hunk 2 above)
Comment (Fadahunsiseyi) [reply spec/views/user_index_spec.rb:33]: @Whoistolu  Thank you, I have fixed the required changes.

**Review by mateo951 — APPROVED** [review]
### Hi Team, 

Well done making the changes requested by the previous reviewer 🚀
Your project is complete!  There is nothing else to say other than... it's time to merge it 🐐 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Fadahunsiseyi** [comment]
@mateo951 Thank you for approving our PR.

**Pull request merged**